Build fails on GCC 11.1 and 11.2 with a format overflow error

On GCC 11.1.0 (tested on ArchLinux) and GCC 11.2.1 (tested on Fedora 34), the build fails with the following message:

```
cc -std=c99 -O3 -Wall -Werror -g -I/usr/include/SDL2 -D_REENTRANT -Isrc/extern/include -Isrc/extern/src -c src/debugger.c -o build/debugger.o
src/debugger.c: In function ‘DEBUGRenderDisplay’:
src/debugger.c:523:48: error: ‘%d’ directive writing between 2 and 10 bytes into a region of size 5 [-Werror=format-overflow=]
  523 |                                 sprintf(lbl, "x%d", reg);
      |                                                ^~
src/debugger.c:523:46: note: directive argument in the range [16, 2147483647]
  523 |                                 sprintf(lbl, "x%d", reg);
      |                                              ^~~~~
src/debugger.c:523:33: note: ‘sprintf’ output between 4 and 12 bytes into a destination of size 6
  523 |                                 sprintf(lbl, "x%d", reg);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:64: build/debugger.o] Error 1
```

Seems to be a false positive given that `reg` is limited to the value of 20 and therefore should never trigger the format overflow.

A simple workaround is to declare `reg` as `unsigned char`.

--- a/src/debugger.c
+++ b/src/debugger.c
@@ -512,7 +512,7 @@ static void DEBUGRenderCmdLine(int x, int width, int height) {
 
 static void DEBUGRenderZeroPageRegisters(int y) {
 #define LAST_R 15
-	int reg = 0;
+	unsigned char reg = 0;
 	int y_start = y;
 	char lbl[6];
 	while (reg < DBGMAX_ZERO_PAGE_REGISTERS) {

